### PR TITLE
[DRAFT] Last value return proposal

### PR DESCRIPTION
**Pull request by airspeedswift**
Adds a rule that the last expression in a function is the return value, and introduces multi-statement `if`, `switch`, and `do` expressions.

### CODE REVIEW

File: proposals/NNNN-last-value-return.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+The introduction of a `then` keyword would set `if` expressions apart from functions. The presence of a `then` keyword in an `if` expression that was part of a single-expression function body would act _like_ a `return` but not exactly, causing cognitive load on the part of the reader.
+
+Overall, the composable consistency between `if`/`switch`/`do` expressions, and function bodies, along with the "`return` means exiting _early_" benefits, weighs in favor of a last expression rule and against a `then` (or similar) keyword.
+
+## Source compatibility
+
+As discussed in detailed design, there are rare edge cases where this new rule may break source, but none have been found in the compatibility test suite. Where they do occur, backticks can be applied, and this fix will back deploy to earlier compiler versions.
```
Comment (benrimmington) [inline proposals/NNNN-last-value-return.md:222]:
```suggestion
As discussed in "Impact on existing code" section, there are edge cases where this new rule may break source.
```
